pan3d/dataset_builder.py

```diff
@@ -69,28 +88,33 @@ def __init__(self, server=None, dataset_path=None, state=None, pangeo=False):
     # -----------------------------------------------------
 
     @property
-    def state(self):
+    def state(self) -> trame_server.state.State:
+        """Returns the current State of the Trame server."""
         return self.server.state
 
     @property
-    def ctrl(self):
+    def ctrl(self) -> trame_server.controller.Controller:
+        """Returns the Controller for the Trame server."""
         return self.server.controller
 
     @property
-    def data_array(self):
+    def data_array(self) -> xarray.core.dataarray.DataArray:
+        """Returns the current Xarray data array with current slicing applied."""
         return self.algorithm.sliced_data_array
 
     @property
-    def mesh(self):
+    def mesh(self) -> pyvista.core.grid.RectilinearGrid:
```

Comment:
I thought it could also be a vtkImageData in some case?

Reply:
Oh right, I'd forgotten can also be a pyvista `StructuredGrid`, so I'll add that. But I don't know if `vtkImageData` can be returned.

Reply:
Added in [aa5dbc8](https://github.com/Kitware/pan3d/pull/47/commits/aa5dbc87edb31c90529218b433aa876d09299801)